tensorflow_federated/python/core/impl/executors/federating_executor.py

```diff
@@ -126,6 +127,318 @@ def type_signature(self):
                                   py_typecheck.type_string(type(self._value))))
 
 
+class IntrinsicStrategy(abc.ABC):
+  @classmethod
+  @abc.abstractmethod
+  def validate_executor_placements(cls, executor_placements):
+    pass
+
+  @classmethod
+  def _check_arg_is_anonymous_tuple(cls, arg):
+    py_typecheck.check_type(arg.type_signature,
+                            computation_types.NamedTupleType)
+    py_typecheck.check_type(arg.internal_representation,
+                            anonymous_tuple.AnonymousTuple)
+
+  @tracing.trace
+  async def _place(self, arg, placement):
+    py_typecheck.check_type(placement, placement_literals.PlacementLiteral)
+    children = self.executor._target_executors[placement]
+    val = await arg.compute()
+    return FederatingExecutorValue(
+        await asyncio.gather(
+            *[c.create_value(val, arg.type_signature) for c in children]),
+        computation_types.FederatedType(
+            arg.type_signature, placement, all_equal=True))
+
+  @tracing.trace
+  async def _eval(self, arg, placement, all_equal):
+    py_typecheck.check_type(arg.type_signature, computation_types.FunctionType)
+    py_typecheck.check_none(arg.type_signature.parameter)
+    py_typecheck.check_type(arg.internal_representation, pb.Computation)
+    py_typecheck.check_type(placement, placement_literals.PlacementLiteral)
+    fn = arg.internal_representation
+    fn_type = arg.type_signature
+    children = self.executor._target_executors[placement]
```

Comment:
rename `executor` to `federating_executor` just to make it super obvious?

Reply:
perhaps define a ctor on `IntrinsicStrategy` that accepts a `FederatingExecutor` and stores it on `self`. right now there's an implicit contract that all derived classes must do this.

Reply:
if you want to push it, you could all make the contract on `self.executor._target_executors` more explicitly, eg by adding `get_target_executors(self, placement)` on `IntrinsicStrategy` (which then does `return self.executor._target_executors(placement)`)